src/main/java/org/takes/rq/CapInputStream.java

```diff
@@ -60,10 +59,7 @@
 
     @Override
     public int available() throws IOException {
-        return (int) Math.min(
-            (long) Integer.MAX_VALUE,
-            Math.max((long) this.origin.available(), this.more)
-        );
+        return (int) Math.min(this.origin.available(), this.more);
```

Comment:
@essobedo yes, it changes the behaviour. This class is used in [RqLengthAware](https://github.com/yegor256/takes/blob/master/src/main/java/org/takes/rq/RqLengthAware.java) that is supposed to respect the Content-Length header. Using the max() between available bytes and `more` value, one can read beyond the Content-Length mark.